junit-jupiter-api/src/main/java/org/junit/jupiter/api/TestReporter.java

```diff
@@ -77,4 +82,34 @@ default void publishEntry(String value) {
 		this.publishEntry("value", value);
 	}
 
+	/**
+	 * Publish the supplied file and attach it to the current test or container.
+	 * <p>
+	 * The file will be copied to the report output directory.
+	 *
+	 * @param file the file to be attached; never {@code null} or blank
+	 * @since 5.11
+	 */
+	@API(status = EXPERIMENTAL, since = "5.11")
+	default void publishFile(Path file) {
+		publishFile(file.getFileName().toString(), path -> Files.copy(file, path, REPLACE_EXISTING));
+	}
+
+	/**
+	 * Publish a file with the supplied name written by the supplied action and
+	 * attach it to the current test or container.
+	 * <p>
+	 * The file will be created in the report output directory prior to invoking
```

Comment:
Is this true?  In the motivating example above, the path is used to create a file or folder in the action lambda.

Reply:
You're right, this isn't true. Clarified in 6256b39a65ac70a2f6b392c213f8bdce78249e51